test/alternative_dialog_spec.rb

```diff
@@ -0,0 +1,91 @@
+require_relative "spec_helper.rb"
+require "update-alternatives/UI/alternative_dialog"
+require "update-alternatives/model/alternative"
+
+describe UpdateAlternatives::AlternativeDialog do
+  def mock_ui_events(*events)
+    allow(Yast::UI).to receive(:UserInput).and_return(*events)
+  end
+
+  def simulate_the_change_of_selected_choice
+    allow(Yast::UI).to receive(:QueryWidget).with(:choices_table, :CurrentItem)
+      .and_return(alternative.value, "/usr/bin/vim")
+  end
+
+  before do
+    allow(Yast::UI).to receive(:OpenDialog).and_return(true)
+    allow(Yast::UI).to receive(:CloseDialog).and_return(true)
+    allow(Yast::UI).to receive(:QueryWidget).with(:choices_table, :CurrentItem)
+      .and_return(alternative.value)
+  end
+
+  subject(:alternative) do
+    UpdateAlternatives::Alternative.new(
```

Comment:
This is not the subject of the test (you are testing the dialog, not an `Alternative` object). Use `let` instead of `subject`.